Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2320,8 +2320,24 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         guard FeatureFlags.current.isExperimentalOverloadedSymbolPresentationEnabled else {
             return
         }
-        
-        try linkResolver.traverseOverloadedSymbols { overloadedSymbolReferences in
+
+        try linkResolver.traverseOverloadedSymbols { (parent, overloadedSymbolReferences) in
+            guard overloadedSymbolReferences.count > 1,
+                let firstOverloadReference = overloadedSymbolReferences.first,
+                let firstOverloadTopicNode = topicGraph.nodeWithReference(firstOverloadReference),
+                let firstOverloadDocumentationNode = try? entity(with: firstOverloadReference),
+                let firstOverloadSymbol = firstOverloadDocumentationNode.semantic as? Symbol else {
+                return
+            }
+
+            // Create the overload group topic reference without a disambiguating hash at the end.
+            guard let indexOfHash = firstOverloadReference.path.lastIndex(of: "-") else { return }
+            let overloadGroupPath = String(firstOverloadReference.path[..<indexOfHash])
+            let overloadGroupReference = ResolvedTopicReference(
+                bundleIdentifier: firstOverloadTopicNode.reference.bundleIdentifier,
+                path: overloadGroupPath,
+                sourceLanguages: firstOverloadTopicNode.reference.sourceLanguages
+            )
```

Comment:
This doesn't guarantee that the reference is unambiguous. In general, directly initializing a `ResolvedTopicReference` from a manually manipulated path is not recommended. 

The place that knows how to create minimal unambiguous references is the link resolver (`PathHierarchy`). If the overload page needs its own unambiguous reference then the path hierarchy needs to be made aware of that symbol before `DocumentationContext.registerSymbols(from:symbolGraphLoader:documentationExtensions:)` is called.

Reply:
Are you suggesting that we move the overload grouping logic into the `PathHierarchy` initializer? That seems like the most likely place to perform that aggregation if the hierarchy is expected to be immutable once all the symbols are registered.

Reply:
Possibly. I _could_ happen after initialization as long as it happens before `DocumentationContext.registerSymbols(from:symbolGraphLoader:documentationExtensions:)`. The logic that identifies the overloads is already in the path hierarchy so making it responsible for creating the node seems fine.

The hierarchy isn't really immutable but symbols shouldn't be added after initialization. Non-symbols don't get their references from the path hierarchy which is the cause of https://github.com/apple/swift-docc/issues/593. 

Also, the path hierarchy shouldn't be manipulating symbols to update the declaration token. That is not its responsibility and you wouldn't retrieve the symbol from the hierarchy anyway, just the unique identifier which maps to a `ResolvedTopicReference`.